workspaces/cli-server/src/routers/capture-router.ts

```diff
@@ -99,6 +94,75 @@ export function makeRouter(dependencies: ICaptureRouterDependencies) {
     }
   );
 
+  ////////////////////////////////////////////////////////////////////////////////
+  router.post(
+    '/initial-bodies',
+    bodyParser.json({ limit: '100mb' }),
+    async (req, res) => {
+      const { captureId } = req.params;
+      const { events, pathId, method } = req.body;
+
+      const initialBodyGenerator = new OnDemandInitialBody({
+        captureBaseDirectory: req.optic.paths.capturesPath,
+        events: events,
+        captureId,
+        pathId,
+        method,
+      });
+
+      await initialBodyGenerator.start();
+
+      const result: Promise<ILearnedBodies> = new Promise((resolve, reject) => {
+        initialBodyGenerator.events.on('progress', (data: any) => {
+          console.log('progress tick ', JSON.stringify(data));
+          if (!data.hasMoreInteractions && data.results) {
+            resolve(data.results);
+          }
+        });
+        initialBodyGenerator.events.on('exit', reject);
+        initialBodyGenerator.events.on('error', reject);
+      });
+
+      result.finally(() => initialBodyGenerator.stop());
```

Comment:
This result Promise is going to leak the EventEmitter and possibly cause a race condition in rejecting a Promise that was already resolved.

I imagine it was introduced as to fit the `start` method of the `Diff` interface, which doesn't allow returning results directly. Since that interface is no longer implemented, I can imagine going back to that. If that poses some issue, perhaps using the `initialBodyGenerator.progress()` steam is a reasonable alternative, which does all the error-prone subscribing and unsubscribing from events, dealing with back-pressure, etc.